validphys2/src/validphys/commondataparser.py

```diff
@@ -1,18 +1,319 @@
 from validphys.coredata import CommonData
 
+EXT = "pineappl.lz4"
+
+
+# Auxiliary parser for common types or sanity checks
+@Parser
+def ValidPath(path_str: str) -> Path:
+    """Parse strings into paths"""
+    return Path(path_str)
+
+
+@Parser
+def ValidOperation(op_str: str) -> str:
+    """Ensures that the operation defined in the commondata file is implemented in validphys"""
+    ret = op_str.upper()
+    # TODO: move accepted operations to this module so that the convolution receives an operation to apply
+    # instead of an operation to understand
+    from validphys.convolution import OP
+
+    if ret not in OP:
+        raise ValidationError(f"The operation '{op_str}' is not implemented in validphys")
+    return ret
+
+
+# Auxiliary objects
+@dataclass
+class ApfelComb:
+    """Some of the grids might have been converted from apfelcomb and introduce hacks.
+    These are the allowed hacks:
+        - repetition_flag:
+            list of fktables which might need to be repeated
+            necessary to apply c-factors in compound observables
+        - normalization:
+            mapping with the single fktables which need to be normalized and the factor
+            note that when they are global factors they are promoted to conversion_factor
+        - shifts:
+            mapping with the single fktables and their respective shifts
+            necessary to create "gaps" that some cfactors or datasets might expect
+    """
+
+    repetition_flag: Optional[list[str]] = None
+    normalization: Optional[dict] = None
+    shifts: Optional[dict] = None
+
+    @classmethod
+    def parser(cls, meta: dict):
+        return parse_input(meta, cls)
+
+ValidApfelComb = Parser(ApfelComb.parser)
+
+@dataclass
+class TheoryMeta:
+    """Contains the necessary information to load the associated fktables"""
+
+    FK_tables: list[list]
+    operation: ValidOperation
+    conversion_factor: float = 1.0 
+    comment: Optional[str] = None
+    apfelcomb: Optional[ValidApfelComb] = None
+    # The following options are transitional so that the yamldb can be used from the theory
+    appl: Optional[bool] = False
+    target_dataset: Optional[str] = None
+
+    def fktables_to_paths(self, grids_folder):
+        """Given a source for pineappl grids, constructs the lists of fktables
+        to be loaded"""
+        ret = []
+        for operand in self.FK_tables:
+            ret.append([grids_folder / f"{m}.{EXT}" for m in operand])
+        return ret
+    
+    @classmethod
+    def parser(cls, meta: dict):
+        """The yaml databases in the server use "operands" instead of "FK_tables" """
+        if "operands" in meta:
+            meta["FK_tables"] = meta.pop("operands")
+        return parse_input(meta, cls)
+
+
+@dataclass
+class KinematicsMeta:
+    """Contains all metadata for the kinematics of the dataset"""
+
+    file: ValidPath
+    variables: dict
+
+    @classmethod
+    def parser(cls, meta: dict):
```

Comment:
not sure why do we need these methods: validobj already behaves like this without them.